functions/src/project/delete-project.ts

```diff
@@ -3,67 +3,67 @@ import { firestore, CloudFunction, EventContext } from 'firebase-functions';
 
 // Dashboard hub firebase functions models/mappers
 import { Logger } from '../client/logger';
-import { ProjectModel } from '../models/index.model';
+import { ProjectModel, RepositoryModel } from '../models/index.model';
 import { deleteMonitorPings } from '../monitor/monitor';
-import { DocumentData, DocumentSnapshot, FirebaseAdmin, WriteResult } from './../client/firebase-admin';
+import { DocumentData, DocumentReference, DocumentSnapshot, FirebaseAdmin, Transaction, WriteResult } from './../client/firebase-admin';
 
-export const onDeleteProjectRepositories: CloudFunction<DocumentSnapshot> = firestore
-  .document('projects/{projectUid}')
-  .onDelete(async (projectSnapshot: DocumentSnapshot, context: EventContext) => {
-
-    try {
-      const project: DocumentData = projectSnapshot.data();
-
-      if (Array.isArray(project.repositories) && project.repositories.length > 0) {
-
-        const promiseList: Promise<WriteResult>[] = [];
-
-        for (const repositoryUid of project.repositories) {
-          const repoData: DocumentData = (await FirebaseAdmin.firestore().collection('repositories').doc(repositoryUid).get()).data();
-
-          if (Array.isArray(repoData.projects)) {
-            repoData.projects = repoData.projects.filter((element: string) => element !== context.params.projectUid);
-          } else {
-            repoData.projects = [];
-          }
-
-          const repo: Promise<WriteResult> = FirebaseAdmin.firestore().collection('repositories').doc(repositoryUid).update(repoData);
-          promiseList.push(repo);
+async function deleteProjectRepositories(projectUid: string, project: ProjectModel): Promise<void> {
+  Logger.info('deleteProjectRepositories');
+  try {
+    if (Array.isArray(project.repositories) && project.repositories.length > 0) {
+      for (const repositoryUid of project.repositories) {
+        if (!repositoryUid) {
+          continue;
         }
+        const repoRef: DocumentReference = RepositoryModel.getRepositoryReference(repositoryUid);
 
-        await Promise.all(promiseList);
-      }
+        await FirebaseAdmin.firestore().runTransaction((t: Transaction) => {
```

Comment:
Missing `return`, the function return type is expecting `promise` return. I will refactor

Reply:
It is not required. This function return Promise<void>